rotkehlchen/db/filtering.py

```diff
@@ -832,8 +853,22 @@ def make(
                 operator='NOT IN',
             ))
         if location is not None:
-            filter_query.location_filter = DBLocationFilter(and_op=True, location=location)
-            filters.append(filter_query.location_filter)
+            if isinstance(location, IncludeExcludeSingleLocationFilterData):
+                if location.operator == '=':
+                    filter_query.location_filter = DBLocationFilter(
+                        and_op=True,
+                        location=location.value,
+                    )
+                    filters.append(filter_query.location_filter)
+                else:  # operator == '!='
+                    filters.append(DBNotEqualFilter(
```

Comment:
there's a bug here due to how `DBLocationFilter` works. 
it sets a `location` property that is used by `get_customized_group_identifiers` so that needs to be handled.